Src/HALALMock/Services/SharedMemory/SharedMemory.cpp

```diff
@@ -94,7 +99,29 @@ void SharedMemory::start_state_machine_memory(){
 	state_machine_count=&state_machine_memory[0];
 	*state_machine_count=0;
 }
-
+void SharedMemory::close(){
+	close_gpio_shared_memory();
+}
+void SharedMemory::close_gpio_shared_memory(){
+	if (gpio_memory != nullptr){
+		//unmap shared memory
+		if(munmap(gpio_memory,state_machine_memory_size) == -1){
+			std::cout<<"Error unmapping the gpio shared_memory\n";
+		}
+		//put the pointer to null
+		gpio_memory = nullptr;
+	}
+	int delete_shared_gpio_memory = shm_unlink(gpio_memory_name);
+	//check the error
+	if(delete_shared_gpio_memory == EACCES){
+		std::cout<<"Permission is denied to unlink the named shared memory object\n";
+	}else if (delete_shared_gpio_memory == ENOENT){
+		std::cout<<"The named shared memory object does not exist\n";
+	}
+	if(close(shm_gpio_fd) == -1){
+		std::cout<<"Error closing the file descriptor\n"
+	}
```

Comment:
```suggestion
	if(shm_gpio_fd!=-1 && close(shm_gpio_fd) == -1){
		std::cout<<"Error closing the file descriptor\n"
		std::terminate();
	}
```
Better to check if the file descriptor is opened